api/src/modules/impact/actual-vs-scenario.service.ts

```diff
@@ -0,0 +1,627 @@
+        ),
+      );
+    });
+    for (const [valueIndex, ImpactTableRowsValues] of entity.values.entries()) {
+      valuesToAggregate.forEach((item: ImpactTableRowsValues[]) => {
+        entity.values[valueIndex].value =
+          ImpactTableRowsValues.value + item[valueIndex].value;
+        entity.values[valueIndex].isProjected =
+          item[valueIndex].isProjected || entity.values[valueIndex].isProjected;
+        if (scenarioId) {
+          entity.values[valueIndex].scenarioValue =
+            (ImpactTableRowsValues.scenarioValue ?? 0) +
+            (item[valueIndex].scenarioValue || 0);
+          entity.values[valueIndex].absoluteDifference =
+            (entity.values[valueIndex].scenarioValue || 0) -
+            entity.values[valueIndex].value;
+          entity.values[valueIndex].percentageDifference =
+            (((entity.values[valueIndex].scenarioValue || 0) -
+              entity.values[valueIndex].value) /
+              entity.values[valueIndex].value) *
+            100;
+        }
+      });
+    }
+    return entity.values;
+  }
+
+  private buildImpactTableRowsSkeleton(
+    entities: ImpactTableEntityType[],
+  ): ImpactTableRows[] {
+    return entities.map((item: ImpactTableEntityType) => {
+      return {
+        name: item.name || '',
+        children:
+          item.children.length > 0
+            ? this.buildImpactTableRowsSkeleton(item.children)
+            : [],
+        values: [],
+      };
+    });
+  }
+
+  private getIdsFromTree(
+    entities: ImpactTableEntityType[],
+    entityIds?: string[],
+  ): string[] {
+    const idsFromTree: string[] = entities.reduce(
+      (ids: string[], entity: ImpactTableEntityType) => {
+        const childIds: string[] =
+          entity.children.length > 0
+            ? this.getIdsFromTree(entity.children)
+            : [];
+        return [...ids, ...childIds, entity.id];
+      },
+      [],
+    );
+    return entityIds
+      ? entityIds.filter((value: string) => idsFromTree.includes(value))
+      : idsFromTree;
+  }
+
+  private static paginateRootEntities(
+    entities: ImpactTableEntityType[],
+    fetchSpecification: FetchSpecification,
+  ): PaginatedEntitiesDto {
+    if (fetchSpecification.disablePagination) {
+      return {
+        entities,
+        metadata: undefined,
+      };
+    }
+    const totalItems: number = entities.length;
+    const pageSize: number =
+      fetchSpecification?.pageSize ?? DEFAULT_PAGINATION.pageSize ?? 25;
+    const page: number =
+      fetchSpecification?.pageNumber ?? DEFAULT_PAGINATION.pageNumber ?? 1;
+    return {
+      entities: entities.slice((page - 1) * pageSize, page * pageSize),
+      metadata: new PaginationMeta({
+        totalPages: Math.ceil(totalItems / pageSize),
+        totalItems,
+        size: pageSize,
+        page,
+      }),
+    };
+  }
+
+  private static processDataForComparison(
+    dataForActualVsScenarioImpactTable: ActualVsScenarioImpactTableData[],
```

Comment:
here data me receive should be of type ImpactTableData[], and the return result of type ActualVsScenarioImpactTableData[]